proxy/proxy/broker/rhnBroker.py

```diff
@@ -686,9 +686,15 @@ def __checkAuthSessionTokenCache(self, token, channel):
         log_debug(4, "Auth channels: '%s'" % authChannels)
         # Check the authorization
         if channel not in authChannels:
-            log_debug(4, "Not subscribed to channel %s; unauthorized" %
-                      channel)
-            raise rhnFault(35, _('Unauthorized channel access requested.'))
+            # Ask for a new token in case we missed channels in the cache
+            cachedToken = self.proxyAuth.update_client_token_if_valid(self.clientServerId, token)
```

Comment:
ok, but this will update the token again and again in case the channel is really not subscribed, right?
I wonder why this was never a problem before?

Reply:
yes sadly, that is the case... and I don't have the answer to your second question sadly

Reply:
In same "painful questions" category: why does it work when we change channels directly, and not when we use SSM?

Reply:
That one is not always true. I have seen at least one case where it failed even without going through SSM. But yes, I don't have the answer to that one either. There is no reason for it to work when changing channels directly: it's the exact same code that is called.